module/apmazure/blob_test.go

```diff
@@ -0,0 +1,293 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build go1.14
+// +build go1.14
+
+package apmazure
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.elastic.co/apm/apmtest"
+)
+
+func TestBlob(t *testing.T) {
+	retry := azblob.RetryOptions{
+		MaxTries: 1,
+	}
+	po := azblob.PipelineOptions{
+		Retry: retry,
+	}
+	p := azblob.NewPipeline(azblob.NewAnonymousCredential(), po)
+	p = WrapPipeline(p)
+	u, err := url.Parse("https://fakeaccnt.blob.core.windows.net")
+	require.NoError(t, err)
+	serviceURL := azblob.NewServiceURL(*u, p)
+	containerURL := serviceURL.NewContainerURL("mycontainer")
+	blobURL := containerURL.NewBlobURL("readme.txt")
+
+	_, spans, errors := apmtest.WithTransaction(func(ctx context.Context) {
+		blobURL.GetTags(ctx, nil)
+	})
+	require.Len(t, errors, 1)
+	require.Len(t, spans, 1)
+	span := spans[0]
+
+	assert.Equal(t, "storage", span.Type)
+	assert.Equal(t, "AzureBlob GetTags mycontainer/readme.txt", span.Name)
+	// TODO: If we use a fake URL, the test is fast but we do not set a
+	// status code
```

Comment:
why don't we set a status code? can we not use an httptest server?

Reply:
Because the request errors, `resp.Response()` is nil, so we can't get a status code. We could use something like `example.blob.core.windows.net` and get a 4XX response, but then the test takes a while longer (~1sec).

If we use httptest.Server, the URL is a local IP addr, but we rely on the subdomains in the provided url to get the account name and storage type (blob, queue, eg): `https://fakeaccnt.blob.core.windows.net`. We could use an httptest server and check the status code is set correctly, but then we won't be able to parse out (in this case) `fakeaccnt` and `blob` for other span properties.

Reply:
I see. What about setting [PipelineOptions.HTTPSender](https://github.com/Azure/azure-storage-blob-go/blob/da3335192ee9609890f4172c43d0750c65941b6f/azblob/zc_pipeline.go#L21-L22)?

Reply:
good idea!